tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/bindgen/BindgenTestCase.java

```diff
@@ -0,0 +1,115 @@
+import org.ballerinalang.test.context.LogLeecher;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static org.ballerinalang.test.context.LogLeecher.LeecherType.ERROR;
+import static org.ballerinalang.test.packaging.PackerinaTestUtils.deleteFiles;
+
+/**
+ * Integration tests for the bindgen tool.
+ *
+ * @since 1.2.5
+ */
+public class BindgenTestCase extends BaseTest {
+
+    private Path tempProjectsDirectory;
+    private Path projectPath;
+    private BMainInstance balClient;
+
+    @BeforeClass()
+    public void setUp() throws IOException, BallerinaTestException {
+        tempProjectsDirectory = Files.createTempDirectory("bal-test-integration-bindgen-");
+        // copy TestProject to a temp
+        Path originalTestProj1 = Paths.get("src", "test", "resources", "bindgen", "TestProject")
+                .toAbsolutePath();
+        projectPath = tempProjectsDirectory.resolve("module1");
+        copyFolder(originalTestProj1, projectPath);
+        balClient = new BMainInstance(balServer);
+    }
+
+    /**
+     * Generate Ballerina bindings covering multiple scenarios and build the project.
+     *
+     * @throws BallerinaTestException Error when executing the commands.
+     */
+    @Test(description = "Test the bindgen command functionality.")
+    public void bindgenTest() throws BallerinaTestException {
+        String buildMsg = "target/bin/module1.jar";
+        LogLeecher buildLeecher = new LogLeecher(buildMsg);
+
+        String bindgenMsg = "class could not be generated.";
+        LogLeecher bindgenLeecher = new LogLeecher(bindgenMsg, ERROR);
+
+        String runMsg1 = "Array util tests successful.";
+        String runMsg2 = "Exception handling tests successful.";
+        LogLeecher runLeecher1 = new LogLeecher(runMsg1);
+        LogLeecher runLeecher2 = new LogLeecher(runMsg2);
+
+        String[] args = {"-mvn=org.yaml:snakeyaml:1.25", "-o=./src/module1", "org.yaml.snakeyaml.Yaml",
+                "java.lang.Object", "java.lang.String", "java.lang.Class", "java.lang.StringBuffer",
+                "java.lang.ArithmeticException", "java.lang.AssertionError", "java.lang.StackTraceElement",
+                "java.lang.Character$Subset", "java.lang.Character", "java.io.File", "java.lang.Short",
+                "java.io.FileInputStream", "java.io.OutputStream", "java.io.FileOutputStream", "java.util.HashSet",
+                "java.util.Set", "java.util.Iterator"};
+        try {
+            balClient.runMain("bindgen", args, null, new String[]{},
+                    new LogLeecher[]{bindgenLeecher}, projectPath.toString());
+            throw new BallerinaTestException("Contains classes not generated.");
```

Comment:
Are we trying to capture an error scenario here? 
Why do we need to throw and error here, catch and re-throw? AFAIK, we can use  `bindgenLeecher.waitForText(5000)` here as it works for ERROR log leechers as well.